routing.go

```diff
@@ -74,6 +74,11 @@ func (dht *IpfsDHT) PutValue(ctx context.Context, key string, value []byte, opts
 		return err
 	}
 
+	// refresh the k-bucket containing this key
+	defer func() {
+		dht.routingTable.BucketForID(kb.ConvertKey(key)).ResetRefreshedAt(time.Now())
```

Comment:
Let's just move this into `GetClosestPeers`. Also, we should probably check if the query succeeded.

Reply:
(or, if it didn't succeed, that it lasted more than... 30 seconds?)